mreg_cli/api/models.py

```diff
@@ -1862,21 +1862,19 @@ class IPAddress(FrozenModelWithTimestamps, WithHost, APIMixin):
 
     @field_validator("macaddress", mode="before")
     @classmethod
-    def create_valid_macadress_or_none(cls, v: str) -> MACAddressField | None:
+    def create_valid_macadress_or_none(cls, v: Any) -> MACAddressField | None:
         """Create macaddress or convert empty strings to None."""
         if v:
             return MACAddressField(address=v)
-
         return None
 
-    @model_validator(mode="before")
+    @field_validator("ipaddress", mode="before")
     @classmethod
-    def convert_ip_address(cls, values: Any):
-        """Convert ipaddress string to IPAddressField if necessary."""
-        ip_address = values.get("ipaddress")
-        if isinstance(ip_address, str):
-            values["ipaddress"] = {"address": ip_address}
-        return values
+    def create_valid_ipaddress(cls, v: Any) -> IPAddressField:
+        """Create macaddress or convert empty strings to None."""
+        if isinstance(v, str):
+            return IPAddressField.from_string(v)
+        return v  # let Pydantic handle it
```

Comment:
A field validator was a better fit here, since we can delegate the task to `IPAddressField.from_string()` directly instead of constructing an intermediate dict representation of the address.